### PR TITLE
PRMP-454 06 NACK tests

### PR DESCRIPTION
**Pull request by bethany-kish-nhs**
- split existing repository E2E into multiple files, only new additions are in RepositoryNackTests
- Removed ehrOut DB config
- Added tests for NACK 06 messages

### CODE REVIEW

File: src/test/java/uk/nhs/prm/e2etests/test/repositorye2etest/RepositoryNackTests.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        final SqsMessage outboundMessage = gp2gpMessengerOQ.getMessageContaining(outboundConversationId);
+
+        //assert the error code is correct
+        assertTrue(outboundMessage.contains("code=\\\"" + nackCode + "\\\""));
+
+        //assert the type code is correct dependent on error code
+        if(nackCode.equals("99")){
+            assertTrue(outboundMessage.contains(unknownNackTypeCode));
+        }
+        else assertTrue(outboundMessage.contains(knownNackTypeCode));
```
Comment (chrisbloe) [inline src/test/java/uk/nhs/prm/e2etests/test/repositorye2etest/RepositoryNackTests.java:114]: if (nackCode.equals("99")) {
            assertTrue(outboundMessage.contains(unknownNackTypeCode));
        } else {
            assertTrue(outboundMessage.contains(knownNackTypeCode));
        }

File: src/test/java/uk/nhs/prm/e2etests/test/repositorye2etest/RepositoryNackTests.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        final String nonExistentNhsNumber = "9729999999";
+
+        //Given we have a message with an unrecognised NHS number
+        EhrRequestTemplateContext ehrRequestContext = EhrRequestTemplateContext.builder()
+                .outboundConversationId(outboundConversationId)
+                .nhsNumber(nonExistentNhsNumber)
+                .senderOdsCode(senderOdsCode)
+                .build();
+
+        String erroneousInboundMessage = this.templatingService.getTemplatedString(EHR_REQUEST, ehrRequestContext);
```
Comment (chrisbloe) [inline src/test/java/uk/nhs/prm/e2etests/test/repositorye2etest/RepositoryNackTests.java:128]: `this.` not required (x3)

**Review by chrisbloe — CHANGES_REQUESTED** [review]
Couple of minor comments.

**Review by chrisbloe-nhse — APPROVED** [review]
Happy with the splitting out of the new tests, and version upgrades.

**Pull request merged**